Chummer/frmNotes.cs

```diff
@@ -107,11 +107,13 @@ private void btnCancel_Click(object sender, EventArgs e)
 
         private void btnColorSelect_Click(object sender, EventArgs e)
         {
-            colorDialog1.Color = ColorManager.GenerateCurrentModeColor(_colNotes);
+            //colorDialog1.Color = ColorManager.GenerateCurrentModeColor(_colNotes);
+            _colNotes = colorDialog1.Color; //Selected color is always how it is shown in light mode, use the stored one for it.
             var resNewColor = colorDialog1.ShowDialog();
             if (resNewColor == DialogResult.OK)
             {
-                _colNotes = ColorManager.GenerateModeIndependentColor(colorDialog1.Color);
+                //_colNotes = ColorManager.GenerateModeIndependentColor(colorDialog1.Color);
+                _colNotes = colorDialog1.Color; //Selected color is always how it is shown in light mode.
```

Comment:
This brings me back to the discussion for:
`Color r = ColorManager.GenerateInverseDarkModeColor(Color.Red);`

I think the following use case might be a bit confusing for a player:
The player has configured DarkMode. In the ColorSelection he selects White as the background color.
I store White as the version for Light Mode and actually show it as Black in Dark Mode.

--> I would think we require a function that is taking a Color that was selected in DarkMode and get back the related Color in LightMode.

Reply:
The idea you went with here is what I originally imagined would work:

* If the user is in Light Mode, the color they picked is stored directly as the color to be picked. When the color is shown, the original color is shown (also via `updateColorRepresentation`) in Light Mode and a generated Dark Mode version is shown in Dark Mode.
* If the user is in Dark Mode, the color they picked is assumed to be geared for Dark Mode. It first gets fed through `ColorManager.GenerateInverseDarkModeColor` to get the color in Light Mode that would turn into the selected one when in Dark Mode and store that. When the color is shown (also via `updateColorRepresentation`), the generated Light Mode color gets shown in Light Mode, while in Dark Mode, the generated Light Mode color has a Dark Mode version generated from it that gets shown. This generated Dark Mode version should end up being either exactly the same color as what the user originally picked in Dark Mode or very close to it.

Does this setup not work properly?

Reply:
Ok, seems like we are aligned how it _should_ work. Therefore I have again commented in line 115 and removed 116.
For LightMode everything is easy, and I just need to store whatever was picked.

Now let us focus on DarkMode, because there it does not work as expected:
* User opens a color picker and selects Red (excuse my German Windows configuration)
![grafik](https://user-images.githubusercontent.com/28984368/119401070-49fb5580-bcdb-11eb-8fa5-eb91c2cff785.png)

* colorDialog1.Color is now set to ARGB(255, 255, 0, 0); `#ff0000`
* ColorManager.GenerateModeIndependentColor calls GenerateInverseDarkModeColor (because we are in DarkMode)
* GenerateInverseDarkModeColor() returns ARGB(255, 255, 255, 255) `#ffffff`
* I store this LightMode version internally, which is already the wrong color
* updateColorRepresentation now calculates DarkMode version from `#ffffff` which turns out to be ARGB (255, 64, 64, 64) `#404040`
* The text is not visible in dark mode:
![grafik](https://user-images.githubusercontent.com/28984368/119401659-0fde8380-bcdc-11eb-911f-34151854d7ed.png)
![grafik](https://user-images.githubusercontent.com/28984368/119401705-25ec4400-bcdc-11eb-92dd-4373f8f557b6.png)


--> GenerateInverseDarkModeColor seems to require changes. It also needs to accept colors like `#ff0000`, `#00ff00`, `#0000ff`